admin/app/routes/authentication/login-oidc.js

```diff
@@ -0,0 +1,110 @@
+  @service router;
+  @service location;
+  @service oidcIdentityProviders;
+
+  beforeModel(transition) {
+    const queryParams = transition.to.queryParams;
+    if (!queryParams.code) {
+      this._cleanSession();
+
+      const identityProviderSlug = transition.to.params.identity_provider_slug.toString();
+      const identityProvider = this.oidcIdentityProviders.list.find((provider) => provider.id === identityProviderSlug);
+      if (identityProvider !== undefined) return this._handleRedirectRequest(identityProvider);
+
+      return this.router.replaceWith('login');
+    }
+  }
+
+  async model(params, transition) {
+    await this.oidcIdentityProviders.loadReadyIdentityProviders();
+
+    const queryParams = transition.to.queryParams;
+    const identityProviderSlug = params.identity_provider_slug;
+    if (queryParams.code) {
+      return this._handleCallbackRequest(queryParams.code, queryParams.state, identityProviderSlug);
+    }
+  }
+
+  async afterModel({ shouldUserCreateAnAccount, authenticationKey, identityProviderSlug, email } = {}) {
+    if (!shouldUserCreateAnAccount && !authenticationKey) return;
+    try {
+      await this.session.authenticate('authenticator:oidc', {
+        authenticationKey,
+        identityProviderSlug,
+        email,
+      });
+    } catch (response) {
+      const apiError = get(response, 'errors[0]');
+      const error = new JSONApiError(apiError?.detail, apiError);
+
+      let queryParams;
+
+      if (error.status === '404' && error.code === 'USER_ACCOUNT_NOT_FOUND') {
+        queryParams = { isUserShouldCreateAnAccount: true };
+      } else {
+        queryParams = { isUnknownErrorOccurs: true };
+      }
+
+      return this.router.replaceWith('login', {
+        queryParams,
+      });
+    }
+  }
+
+  _cleanSession() {
+    this.session.set('data.nextURL', undefined);
+  }
+
+  async _handleCallbackRequest(code, state, identityProviderSlug) {
+    try {
+      const redirectUri = this._getRedirectUri(identityProviderSlug);
+      await this.session.authenticate('authenticator:oidc', {
+        code,
+        redirectUri,
+        state,
+        identityProviderSlug,
+      });
+    } catch (response) {
+      const apiError = get(response, 'errors[0]');
+      const error = new JSONApiError(apiError.detail, apiError);
+
+      const shouldUserCreateAnAccount = error.code === 'SHOULD_VALIDATE_CGU';
+      const { authenticationKey, email } = error.meta ?? {};
+      if (shouldUserCreateAnAccount && authenticationKey) {
+        return { shouldUserCreateAnAccount, authenticationKey, email, identityProviderSlug };
+      }
+
+      if (error.status === '403' && error.code === 'PIX_ADMIN_ACCESS_NOT_ALLOWED') {
+        return this.router.replaceWith('login', {
+          queryParams: {
+            isUserShouldRequestAccess: true,
+          },
+        });
+      }
+
+      throw error;
+    }
+  }
+
+  _getRedirectUri(identityProviderId) {
+    const { protocol, host } = location;
```

Comment:
Compte tenu qu'il y a le `@service location`, est-ce que préciser `window.location` ne serait pas mieux ? Personnellement en général je précise toujours `window.location`, c'est plus explicite.

```suggestion
    const { protocol, host } = window.location;
```

PS : Je sais qu'il y a le même code dans *mon-pix* et qu'il faudrait le modifier là-bas aussi pour être cohérent. Et je veux bien m'en charger ;-)